src/components/ListManager/ListManager.vue

```diff
@@ -0,0 +1,212 @@
+        } else {
+          for (let i = 0; i < manager.value.list.length; i++) {
+            selectedItems.value[manager.value.list[i].name] = manager.value.list[i]
+          }
+        }
+      },
+      _select: (rowData) => {
+        if (selectionMode.value == 0) {
+          selectionMode.value = 1
+        }
+        if(rowData.name in selectedItems.value) {
+          delete selectedItems.value[rowData.name]
+          if (Object.keys(selectedItems.value).length == 0) {
+            selectionMode.value = 0
+          }
+        } else {
+          selectedItems.value[rowData.name] = rowData
+        }
+      },
+    })
+    const clearList = () => {
+      manager.value.list = []
+      newItems.value = []
+      selectedItems.value = {}
+    }
+    context.expose({ manager })
+    return {
+      manager,
+      newItems,
+      selectedItems,
+      selectionMode
+    }
+  },
+  mounted() {
+    this.manager.resource = this.$resources.list
+    this.handleRealtimeUpdate()
+  },
+  resources: {
+    list() {
+      return {
+        type: 'list',
+        doctype: this.options?.doctype,
+        fields: [...this.options?.fields, "name"],
+        cache: this.options?.cache,
+        order_by: this.options?.order_by,
+        filters: this.options?.filters,
+        start: 0,
+        limit: this.options?.limit || 20,
+        onSuccess: (data) => {
+          /**
+           * Remove all the duplicates which might have been added to the
+           * current list when new entry was added via socket list_update
+           */
+          var newItems = this.newItems
+          data = data.filter(function(i) {
+            return !newItems.find(j => {
+              return j === i.name
+            });
+          });
+          
+          this.manager.list = [...this.manager.list, ...data]
+          this.manager.loading = false
+        }
+      }
+    },
+    document() {
+      return {
+        method: 'frappe.client.get_list',
+        onSuccess: (data) => {
+          if (data.length > 0) {
+            const itemIndex = this.manager.list.findIndex(item => item.name === data[0].name)
+            if (itemIndex != -1) {
+              this.manager.list[itemIndex] = data[0]
+            } else {
+              this.manager.list.unshift(data[0])
+              this.newItems.push(data[0].name)
+            }
+            this.manager.loading = false
+          }
+        }
+      }
+    }
+  },
+  methods: {
+    onScroll({ target: { scrollTop, clientHeight, scrollHeight }}) {
+			if (scrollTop + clientHeight >= scrollHeight) {
+				if (this.manager.options.auto_pagination) {
+          this.manager.loadMore()
+        }
+			}
+		},
+    handleRealtimeUpdate() {
+      this.$socket.on("list_update", (data) => {
+        if (data.doctype === this.options.doctype) {
+          this.$resources.document.fetch({
+            doctype: this.manager.options.doctype,
+            filters: {
+              ...this.options.filters,
```

Comment:
No need for any other filters if `name` filter is set

Reply:
this was done to reduce unwanted API fetches, if the new updated item does not follow the filters passed via the user inside the list manager options, it does not make sense to handle the update either right

also, I think this is how the desk (framework) list UI handles list updates too